components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorAPI.java

```diff
@@ -298,12 +298,12 @@ public Response callback(@Context Request request, @PathParam("appName") String
                 }
                 // Redirect to the store/apis page (redirect URL)
                 URI targetURIForRedirection = new URI(appConfigs.getApimBaseUrl() + appName);
-                if (AuthenticatorConstants.PUBLISHER_APPLICATION.equals(appName)) {
+                if (AuthenticatorConstants.PUBLISHER_APPLICATION.equals(appName) || AuthenticatorConstants.STORE_APPLICATION.equals(appName)) {
                     String authResponseBeanData = authResponseBean.getAuthUser() + "&id_token="
                             + authResponseBean.getIdToken() + "&partial_token=" + authResponseBean.getPartialToken()
                             + "&scopes=" + authResponseBean.getScopes() + "&validity_period="
                             + authResponseBean.getValidityPeriod();
-                    URI redirectURI = new URI(appConfigs.getApimBaseUrl() + "publisher/login?user_name="
+                    URI redirectURI = new URI(appConfigs.getApimBaseUrl() + appName + "/login?user_name="
                             + URLEncoder.encode(authResponseBeanData, "UTF-8")
                             .replaceAll("\\+", "%20").replaceAll("%26", "&")
```

Comment:
Formatting issue. I think '.replaceAll("\\+", "%20").replaceAll("%26", "&")' and '.replaceAll("%3D", "="));' can be in one line as below.
.replaceAll("\\+", "%20").replaceAll("%26", "&").replaceAll("%3D", "="));